.circleci/config.yml

```
@@ -46,6 +46,22 @@ commands:
           root: .
           paths:
             - ./pkg/
+  build-and-persist-packer-binary-arm64:
+    parameters:
+      GOOS:
+        type: string
+      GOARCH:
+        default: "arm64"
+        type: string
+    steps:
+      - checkout
+      - run: GOOS=<< parameters.GOOS >> go build -ldflags="-s -w -X github.com/hashicorp/packer/version.GitCommit=${CIRCLE_SHA1}" -o ./pkg/packer_<< parameters.GOOS >>_<< parameters.GOARCH >> .
```

Looks like the darwin bin is still being built for amd64. We need to set the GOARCH here 

```suggestion
      - run: GOOS=<< parameters.GOOS >> GOARCH=<<parameters.GOARCH>> go build -ldflags="-s -w -X github.com/hashicorp/packer/version.GitCommit=${CIRCLE_SHA1}" -o ./pkg/packer_<< parameters.GOOS >>_<< parameters.GOARCH >> .
```